nucleus/core/kernel/src/main/java/org/glassfish/runnablejar/UberMain.java

```diff
@@ -213,23 +252,38 @@ private void addShutdownHook() {
 
             @Override
             public void run() {
-                if (glassFish != null) {
-                    stopGlassFish();
+                // We restore the handlers because they are already removed from the logger in the shutdown hook
```

Comment:
This is quite certainly bad idea, I will remember later why.
Btw logging should not be used for communication with the user, but we have this problem all over the sources. I already created an issue for that.

Reply:
This is not communication with the user, it’s a proper log message at the end. It can happen in Kubernetes for example, if the service is misconfigured.

what else do you suggest? System.err?